xarray/tests/test_units.py

```diff
@@ -1245,6 +1311,766 @@ def test_dot_dataarray(dtype):
+            strip_units(convert_units(right, units))
+        ) & is_compatible(unit, base_unit)
+        actual = left.broadcast_equals(right)
+
+        assert expected == actual
+
+    @pytest.mark.parametrize(
+        "indices",
+        (
+            pytest.param(4, id="single index"),
+            pytest.param([5, 2, 9, 1], id="multiple indices"),
+        ),
+    )
+    def test_isel(self, indices, dtype):
+        array = np.linspace(0, 5, 10).astype(dtype) * unit_registry.s
+        variable = xr.Variable("x", array)
+
+        expected = attach_units(
+            strip_units(variable).isel(x=indices), extract_units(variable)
+        )
+        actual = variable.isel(x=indices)
+
+        xr.testing.assert_identical(expected, actual)
+
+    @pytest.mark.parametrize(
+        "unit,error",
+        (
+            pytest.param(1, DimensionalityError, id="no_unit"),
+            pytest.param(
+                unit_registry.dimensionless, DimensionalityError, id="dimensionless"
+            ),
+            pytest.param(unit_registry.s, DimensionalityError, id="incompatible_unit"),
+            pytest.param(unit_registry.cm, None, id="compatible_unit"),
+            pytest.param(unit_registry.m, None, id="identical_unit"),
+        ),
+    )
+    @pytest.mark.parametrize(
+        "func",
+        (
+            function(lambda x, *_: +x, function_label="unary_plus"),
+            function(lambda x, *_: -x, function_label="unary_minus"),
+            function(lambda x, *_: abs(x), function_label="absolute"),
+            function(lambda x, y: x + y, function_label="sum"),
+            function(lambda x, y: y + x, function_label="commutative_sum"),
+            function(lambda x, y: x * y, function_label="product"),
+            function(lambda x, y: y * x, function_label="commutative_product"),
+        ),
+        ids=repr,
+    )
+    def test_1d_math(self, func, unit, error, dtype):
+        base_unit = unit_registry.m
+        array = np.arange(5).astype(dtype) * base_unit
+        variable = xr.Variable("x", array)
+
+        values = np.ones(5)
+        y = values * unit
+
+        if error is not None and func.name in ("sum", "commutative_sum"):
+            with pytest.raises(error):
+                func(variable, y)
+
+            return
+
+        units = extract_units(func(array, y))
+        if all(compatible_mappings(units, extract_units(y)).values()):
+            converted_y = convert_units(y, units)
+        else:
+            converted_y = y
+
+        if all(compatible_mappings(units, extract_units(variable)).values()):
+            converted_variable = convert_units(variable, units)
+        else:
+            converted_variable = variable
+
+        expected = attach_units(
+            func(strip_units(converted_variable), strip_units(converted_y)), units
+        )
+        actual = func(variable, y)
+
+        assert extract_units(expected) == extract_units(actual)
```

Comment:
we have `assert_chunks_equal` so it makes sense (to me) that we add `assert_units_compatible`. Thoughts?

Reply:
this is more of a band-aid until `assert_allclose` and `assert_identical` check unit equality (compatibility is not enough for this). It is true, though, that it is used often enough to put it into a function.

Reply:
ah this looks fine

Reply:
I changed it since for new tests (or when I switch the `Dataset` tests to `assert_allclose` / `assert_identical`) I don't have to type as much